src/org/sosy_lab/java_smt/solvers/apron/ApronRationalFormulaManager.java

```diff
@@ -0,0 +1,260 @@
+import org.sosy_lab.java_smt.solvers.apron.types.ApronNode.ApronNumeralNode.ApronRatUnaryNode;
+import org.sosy_lab.java_smt.solvers.apron.types.ApronNode.ApronNumeralNode.ApronRatVarNode;
+
+public class ApronRationalFormulaManager extends
+                                         ApronNumeralFormulaManager<NumeralFormula, RationalFormula>
+    implements RationalFormulaManager {
+
+  private final ApronFormulaCreator apronFormulaCreator;
+  private final ApronFormulaType rationalType = new ApronRationalType();
+
+  protected ApronRationalFormulaManager(
+      ApronFormulaCreator pFormulaCreator,
+      NonLinearArithmetic pNonLinearArithmetic) {
+    super(pFormulaCreator, pNonLinearArithmetic);
+    this.apronFormulaCreator = pFormulaCreator;
+  }
+
+  @Override
+  protected FormulaType getNumeralType() {
+    return FormulaType.RATIONAL;
+  }
+
+  @Override
+  protected ApronNode makeVariableImpl(String i) {
+    return apronFormulaCreator.makeVariable(rationalType, i);
+  }
+
+  @Override
+  protected ApronNode makeNumberImpl(double pNumber) {
+    BigDecimal dec = BigDecimal.valueOf(pNumber);
+    Rational rat = Rational.ofBigDecimal(dec);
+    return new ApronRatCstNode(rat.getNum(), rat.getDen());
+  }
+
+  @Override
+  protected ApronNode makeNumberImpl(BigDecimal pNumber) {
+    Rational rat = Rational.ofBigDecimal(pNumber);
+    return new ApronRatCstNode(rat.getNum(), rat.getDen());
+  }
+
+  @Override
+  protected ApronNode makeNumberImpl(long i) {
+    return new ApronRatCstNode(BigInteger.valueOf(i), BigInteger.ONE);
+  }
+
+  @Override
+  protected ApronNode makeNumberImpl(BigInteger i) {
+    return new ApronRatCstNode(i, BigInteger.ONE);
+  }
+
+  @Override
+  protected ApronNode makeNumberImpl(String i) {
+    Preconditions.checkArgument(!(i.contains(".") || i.contains(",")),
+        "Rational number has to be written like 2/5.");
+    List<String> numbers = Splitter.on('/').splitToList(i);
+    BigInteger num = new BigInteger(numbers.get(0));
+    if (numbers.size() > 1) {
+      BigInteger den = new BigInteger(numbers.get(1));
+      return new ApronRatCstNode(num, den);
+    }
+    return new ApronRatCstNode(num, BigInteger.ONE);
+  }
+
+  @Override
+  protected ApronNode negate(ApronNode pParam1) {
+    ApronRatUnaryNode unaryNode = new ApronRatUnaryNode(pParam1,
+        Texpr1UnNode.OP_NEG);
+    return unaryNode;
+  }
+
+  @Override
+  protected ApronNode add(ApronNode pParam1, ApronNode pParam2) {
+    ApronRatBinaryNode binaryNode = new ApronRatBinaryNode(pParam1, pParam2,
+        Texpr1BinNode.OP_ADD);
+    return binaryNode;
+  }
+
+  @Override
+  protected ApronNode sumImpl(List<ApronNode> operands) {
+    if (!operands.isEmpty()) {
+      ApronNode first = operands.remove(0);
+      for (ApronNode operand : operands) {
+        first = new ApronRatBinaryNode(first, operand,
+            Texpr1BinNode.OP_ADD);
+      }
+      return first;
+    }
+    return null;
```

Comment:
Is `null` the correct return here?
Maybe throw an exception instead?

Reply:
I'm returning 0 now.